src/components/AutoComplete/AutoComplete.js

```diff
@@ -0,0 +1,127 @@
+import React, { Component } from 'react';
+import Downshift from 'downshift';
+import { InputGroup, Button } from '../../index';
+import PropTypes from 'prop-types';
+
+import AutoCompleteInput from './AutoCompleteInput';
+import AutoCompleteItems from './AutoCompleteItems';
+
+class AutoComplete extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      inputValue: ''
+    };
+  }
+
+  render() {
+    const {
+      onSearch,
+      labelText,
+      onInputUpdate,
+      items,
+      actionText,
+      ...rest
+    } = this.props;
+
+    const activeItems = items
+      .filter(
+        ({ disabled, type }) =>
+          !disabled && !['header', 'divider'].includes(type)
+      )
+      .map(({ text }) => text);
+
+    return (
+      <Downshift
+        onStateChange={({ inputValue, ...rest }) => {
+          if (typeof inputValue === 'string') {
+            onInputUpdate(inputValue);
+            this.setState({ inputValue });
+          }
+        }}
+        defaultHighlightedIndex={0}
+        selectedItem={this.state.inputValue}
+        {...rest}
+      >
+        {({
+          getInputProps,
+          getItemProps,
+          getLabelProps,
+          isOpen,
+          inputValue,
+          highlightedIndex,
+          getRootProps,
+          selectedItem,
+          selectItem
+        }) => (
+          <div>
+            {labelText && <label {...getLabelProps()}>{labelText}</label>}
+            <InputGroup>
+              <AutoCompleteInput
+                onKeyPress={e => {
+                  const TAB_KEY = 9;
```

Comment:
maybe we can go ahead and move these to `common/helpers.js` ?

Reply:
I wish there was a way to destructure the properties but I like having them in the common folder.